File: 7_kyu/Number_of_People_in_the_Bus.go
package kata

// Calculates the number of people in the bus.
// busStops for each stop: people get into bus (first item)
//        and number of people get off the bus (second item)
func Number(busStops [][2]int) int {
	number := 0
	for _, it := range busStops {
		number += it[0]
		number -= it[1]
	}
	return number
}

File: 7_kyu/Number_of_People_in_the_Bus.rs
// Returns the number of people on the bus at the last stop.
// Argument is an array of pairs (#people in, #people out) at each stop.
fn number(bus_stops: &[(i32, i32)]) -> i32 {
    bus_stops.iter().map(|in_out| in_out.0 - in_out.1).sum()
}
